roles/download_rootfs_hypershift/tasks/main.yaml

```
@@ -0,0 +1,45 @@
+---
+- name: install apache httpd
+  package:
+    name: httpd
+    state: latest
+
+- name: bind httpd to port 8080 
+  lineinfile:
+    dest: /etc/httpd/conf/httpd.conf
+    regexp: ^Listen 80
+    line: Listen 8080
+
+- name: allow traffic at 8080 for apache 
+  tags: firewall
+  firewalld:
+    port: 8080/tcp
+    zone: "{{ item }}"
+    state: enabled
+    permanent: true
+  with_items:
+  - internal
+  - public
+
+
+- name: Download ipxe script 
+  shell: curl -k -L $(oc -n {{ hypershift.hcp.clusters_namespace }}-{{ hypershift.hcp.hosted_cluster_name }} get InfraEnv {{ hypershift.hcp.hosted_cluster_name }} -ojsonpath="{.status.bootArtifacts.ipxeScript}")
+  register: ipxe_script
+
+- name: Get URL for rootfs
+  set_fact:
+    rootfs_url: "{{ ipxe_script.stdout_lines[2].split(' ')[3].split('url=')[1] }}"
+
+- name: Download rootfs image 
+  shell: curl -k -L -o /var/www/html/rootfs.img "{{ rootfs_url }}"
```

```suggestion
- name: Download rootfs image 
  shell: curl -k -L -o /var/www/html/rootfs.img "{{ rootfs_url }}"
```
Any reason this can't use the get_url module?

---

Updated to use  get_url module